### ISSUE TITLE
[BUG]: PCSX2 MacOS PCAP not connecting

### ISSUE THREAD

**Issue by Leyo1** [PCSX2/pcsx2#7650]
### Describe the Bug

when connecting online using "PCAP Bridged" or "PCAP Switched" the games would display this message: "You must have a network adapter installed in your PlayStation2 to play online", but if you choose "Socket" it would go online normally but can't host games or join them (tested with Syphon Filter: The Omega Strain)

### Reproduction Steps

Tested on PCAP Bridged, PCAP Switched, & Socket
Tested on MacOS Ventura but Ventura is not listed so i chose MacOS Monteray as it's the closest

### Expected Behavior

Expected behavior is for PCAP Bridged and Switched to work normally and for the games to detect them as adapters

### PCSX2 Revision

v1.7.3735

### Operating System

macOS (Monteray)

**Comment by CykaDelik** [comment]
Having this exact same issue. 

SteamOS (Arch Linux)

I have libpcap installed via Homebrew [https://brew.sh/] [https://formulae.brew.sh/formula/libpcap]

Bridged/Switched, PCSX2 can't find the Network adapter

Switching to sockets allows me to find the network adapter and connect to the server and create a lobby. No one can connect to my lobby, and I can't join any other lobbies.
I can see other people's lobbies and they can see mine.

**Comment by TheLastRar** [comment]
Mac not supporting pcap is a known issue

PCAP on Linux should work, but do note the following;
AppImage builds don't work with due to permission issues, you'll need to extract the AppImage with `--appimage-extract`
Flatpack builds don't work with due to permission issues, try the workarounds listed [here](https://github.com/flathub/net.pcsx2.PCSX2#networking-support).

**Comment by lunelovehearn** [comment]
I have the same issue but under Arch Linux. pcap only works with the root account, but not under a regular user. this is a bug and should be fixed as it's not recommended to run everything as root/pcsx2 shouldn't be ran like this. I use the aur and flatpak versions and same issue.

**Comment by weirdbeardgame** [comment]
It's not a bug. There's specific groups your user has to be added to.

**Comment by lunelovehearn** [comment]
> It's not a bug. There's specific groups your user has to be added to.

then, what groups should I be added to?
mind elaborating the solution?

**Comment by F0bes** [comment]
I can't speak for the appimage, but the `netdev` group has fixed permissions for me previously.

**Comment by lunelovehearn** [comment]
> 

that group doesn't exists on my arch system.

**Comment by F0bes** [comment]
Then add it.

**Comment by lunelovehearn** [comment]
> Then add it.

did, then added myself to it then chgrp pcsx2 binary on /usr/bin

**Comment by lunelovehearn** [comment]
any news?

**Comment by TheLastRar** [comment]
> any news?

Can you post an emulog

**Comment by faelana** [comment]
I also have the same issue.

**Comment by faelana** [comment]
@TheLastRar I tried the flatpak configuration but it didnt work for me. 
@CykaDelik did you find a solution?
@weirdbeardgame where can i see which groups should i add?
@F0bes how did you come to netdev?
@lunelovehearn did you find a solution? to me not even on root it works.

**Comment by TheLastRar** [comment]
> @TheLastRar I tried the flatpak configuration but it didn't work for me.

Unfortunately not, PCSX2 has the same issue as wireshark under flatpak, which still doesn't support capture https://github.com/flathub/org.wireshark.Wireshark/issues/4

You will have to unpack the flatpak (or grab the appimage and unpack that) and use setcap to restore the missing capabilities

**Comment by CykaDelik** [comment]
> @TheLastRar I tried the flatpak configuration but it didnt work for me. @CykaDelik did you find a solution? @weirdbeardgame where can i see which groups should i add? @F0bes how did you come to netdev? @lunelovehearn did you find a solution? to me not even on root it works.

Nah. I'm just going to wait for a native solution that doesn't involve any unpacking.
Might be waiting for a while, heh
Would love to play some Twisted Metal, BattleFront or MGS3 online on steam deck someday soon =)

**Comment by faelana** [comment]
> > @TheLastRar I tried the flatpak configuration but it didn't work for me.
> 
> Unfortunately not, PCSX2 has the same issue as wireshark under flatpak, which still doesn't support capture [flathub/org.wireshark.Wireshark#4](https://github.com/flathub/org.wireshark.Wireshark/issues/4)
> 
> You will have to unpack the flatpak (or grab the appimage and unpack that) and use setcap to restore the missing capabilities

./pcsx2-qt: error while loading shared libraries: libSoundTouch.so.1: cannot open shared object file: No such file or directory

is this included in the appimage? i cant launch without it.

**Comment by lunelovehearn** [comment]
> 

no, on arch based on my testing, it works fine under root account, that is, using sudo pcsx2, tho user shouldn't run an emulator with root account. haven't made any more research so far.

**Comment by lunelovehearn** [comment]
> > any news?
> 
> Can you post an emulog

lemme figure out how to, then will post the logs of it under a regular and root user accounts.

**Comment by devmni** [comment]
> Having this exact same issue.
> 
> SteamOS (Arch Linux)
> 
> I have libpcap installed via Homebrew [https://brew.sh/] [https://formulae.brew.sh/formula/libpcap]
> 
> Bridged/Switched, PCSX2 can't find the Network adapter
> 
> Switching to sockets allows me to find the network adapter and connect to the server and create a lobby. No one can connect to my lobby, and I can't join any other lobbies. I can see other people's lobbies and they can see mine.

I have the exact same problem. Connecting to a lobby works fine while using "Sockets", but i can't start or join any session. I tried this for Formula '05, Midnight Club 3, GT4 Online Public Beta and NfS Underground. Just Midnight Club tells me that an "NAT Negotiation error" occured after trying to start a session.

In PCAP it wont find a network adapter (no MAC shown in version info) and as sudo in only finds a network adapter in "PCAP bridged" but say's "No Cable connected".

Did you ever find an workaround for this?

**Comment by HackTime** [comment]
my pcsx2 have the same problem, but i only could make work when i change to "Sockets" (i install my pcsx2 from terminal)

**Comment by madhavpcm** [comment]
Like mentioned above, not running with appimages helps detect the network adaptor in PCAP Bridged and PCAP Switched. This was an issue with the PCSX2 which comes bundled with Lutris, but upon using the [AUR](https://aur.archlinux.org/packages/pcsx2) version, the games are able to detect the network adaptor. However, I am not able to see a dhcp lease of the IP which I set in game in my router.

**Comment by stenzek** [comment]
We have nothing to do with the AUR. Speak to your packager for support.

**Comment by madhavpcm** [comment]
> We have nothing to do with the AUR. Speak to your packager for support.

Sorry for the confusion, but I should have emphasised more on the fact that network adaptor gets detected if you are running natively (like building your self or as a package) it should work.

**Comment by totolgueimer** [comment]
> > It works but cant play online Burnout 3
> 
> I heard that recently Sockets was updated and now works fully, so that's worth a try even if it was tested before, there are new fixes.
> 
> Also for PCAP on Linux, i saw people installing PCSX2 using the Synaptic Package Manager instead of Flatpack/AppImage downloaded from the site, and after that they add the PCAP permissions in Terminal, that made PCAP work on Linux.
> 
> Steps: Add PCSX2 ppa in "Software Sources" or in Terminal: sudo add-apt-repository ppa:pcsx2-team/pcsx2-daily Install PCSX2 from "Synaptic Package Manager" Add PCAP Permissions in Terminal: sudo setcap 'CAP_NET_RAW+eip CAP_NET_ADMIN+eip' /opt/pcsx2-unstable/pcsx2-qt
> 
> As for PCAP on MacOS, that remains unsolved.

i just found out that my pcsx2 installation already had the permissions, and for getting the emulator connected to the internet with PCAP i just had to intercept the DHCP and change the PS2 IP Address

**Comment by totolgueimer** [comment]
but still can't play Burnout 3 because i can't open ports, i know how to, but i have to pay extra to Megacable if i want my own Public IP Address

**Comment by Double-0-seven7** [comment]
Its the most problematic getting PCAP to work on the Steam Deck because the filesystem there is immutable and you cant use something like the AUR in a straightforward way or compile it.
Best way I found is by using Distrobox in root mode but its not that trivial to set up (extracting the appimage didn't work last time I tried).

**Comment by chakaponden** [comment]
> 
> Steps: Add PCSX2 ppa in "Software Sources" or in Terminal: sudo add-apt-repository ppa:pcsx2-team/pcsx2-daily Install PCSX2 from "Synaptic Package Manager" Add PCAP Permissions in Terminal: sudo setcap 'CAP_NET_RAW+eip CAP_NET_ADMIN+eip' /opt/pcsx2-unstable/pcsx2-qt
> 
> As for PCAP on MacOS, that remains unsolved.

Confirm PCSX2 online multiplayer works on Ubuntu 24.04 with installing from pcsx2-daily.
Works with WLAN via Socket & via Pcap with cable connection

**Comment by LoKKeR96** [comment]
> In case it helps anyone, I was having trouble trying to connect with PCSX2 - Mortal Kombat Armageddon, and this was what worked for me (Sockets/Auto):
> 
> <img alt="image" width="786" src="https://private-user-images.githubusercontent.com/897442/296673355-37907caa-e366-414d-b5e5-20bdbc352d41.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ2NDAzMDQsIm5iZiI6MTc0NDY0MDAwNCwicGF0aCI6Ii84OTc0NDIvMjk2NjczMzU1LTM3OTA3Y2FhLWUzNjYtNDE0ZC1iNWU1LTIwYmRiYzM1MmQ0MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxNFQxNDEzMjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MGY1ZmYwODc0MzNkZGQ5YmFjYjIyOWM5ZDBhYzJlN2RmM2FkZGFiOWYwMWI2ODMwNTkwNGI5MzRmNDBiOGJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZJRpSARD18umf7gfxRZyPPLmpK44xNr5l6mhJlS2XDg">
> And with the connection settings within the game itself, I kept everything on Auto as well, but specified the Primary DNS manually.

This worked for me on MacOS Sequoia 15.4 on my Macbook M2 Pro. I'm playing MGS3 online (also known as MGO1).
I can connect to servers and play.

**Comment by jwetzell** [comment]
I can as well confirm that PCAP permissions don't seem to be an issue when installed via `apt` with the `pcsx2-team/pcsx2-daily` repo. Has there been any definitive way to get the AppImage working even extracting and various setcap either make launching not work or still show the permissions errors when launching with PCAP modes.

**Comment by llsant** [comment]
I was about to open a related issue and I found this. Your network configuration may vary, in my case, I'm emulating PS2 LAN games over a VPN. What I found (both on Windows and on Linux) is that the network adapter must have an IPv4 on the host. You can put anything, ideally a link-local or private IP. But if there's no IPv4 in the host, PCSX2 will have trouble even though it shouldn't. In pcap mode, the only IP that matters is whatever you configure inside the game or network access disk, because pcap should just be moving ethernet frames generated by the virtual ps2 to the interface.

A way to check for this problem is going to the log and finding "DEV9" entries. If the app has proper permissions (either Windows or Linux) to open the device, you'll see lines saying

DEV9: Adapter ok

However, after this line, if the interface lacks an IPv4, you'll find a line right after that says something like "Couldn't get MacAddress() info" (very similar, I can't remember exactly). On Windows this is usually less of an issue because VPN TAP interfaces use the OpenVPN driver for the most part, and without any extra configuration, they have a link-local 169.254.0.0/16 address.

So basically, keep your setups but go to ncpa.cpl (network connections control panel), right click on the interface and set an IP. Close and reopen PCSX2. Not only is it necessary to see "Adapter OK" in the logs, but also no messages about MacAdress() error or "unable to get adapter configuration".

Hope this helps. However I'm going to open a ticket because it shouldn't happen, layer 2 is layer 2, the OS IP address shouldn't interfere.
